### ISSUE TITLE
Request to support HTTP_PROXY and NO_PROXY terraform workspace variables

### ISSUE THREAD

**Issue by rayuduc** [fgouteroux/terraform-provider-mimir#16]
Hello,
   I truly appreciate for the Mimir Terraform provider. I'm reaching out with a feature request that I believe would greatly benefit the community as well.

**Use Case:** Currently, we use Grafana Cloud in conjunction with Terraform Enterprise.  We're trying to set up HTTP proxy settings using the HTTP_PROXY and NO_PROXY variables in Terraform workspaces. Currently, it seems Mimir Terraform provider doesn't allow HTTP proxy settings. It would be really helpful if Mimir Terraform provider support HTTP proxy setting.

Here is my provider configuration looks like.
```
terraform {
  required_providers {
    mimir = {
      source  = "fgouteroux/mimir"
      version = "0.1.7"
    }
  }
}
provider "mimir" {
  ruler_uri        = "https://prometheus-us-central1.grafana.net/prometheus"  #test
  alertmanager_uri = "https://alertmanager-us-central1.grafana.net" 
  org_id           =  sandbox_test
  username         = 39909 (prometheus instance ID)
  password         = xxxxxxxxxxx
}
provider "mimir" {
  ruler_uri        = "https://prometheus-us-central1.grafana.net/prometheus" #test
  alertmanager_uri = "https://alertmanager-us-central1.grafana.net"
  org_id           = sandbox_test
  username         = 49588 (alertmanager instance ID)
  password         = xxxxxxxxxxx
}
```

**Comment by fgouteroux** [comment]
Hello @rayuduc,

Implemented [here](https://github.com/fgouteroux/terraform-provider-mimir/blob/main/mimir/api_client.go#L101) and released in [v0.1.8](https://github.com/fgouteroux/terraform-provider-mimir/releases/tag/v0.1.8), you can also use the `proxy_url` (cf: https://registry.terraform.io/providers/fgouteroux/mimir/latest/docs#proxy_url).

**Comment by rayuduc** [comment]
Thanks  a lot @fgouteroux.
